src/odemis/acq/test/move_test.py

```diff
@@ -252,23 +251,27 @@ def test_get_current_position(self):
 
         # Move to imaging position and cancel the movement before reaching there
         f = cryoSwitchSamplePosition(THREE_BEAMS)
-        # abit long wait for the loading-imaging referencing to finish
+        # wait just long enough for the referencing to complete
         time.sleep(7)
         f.cancel()
         pos_label = getCurrentPositionLabel(stage.position.value, stage)
-        self.assertEqual(pos_label, LOADING_PATH)
+        # It's really hard to get the timing right, so also allow to be at loading
+        # or three-beams
+        self.assertIn(pos_label, (THREE_BEAMS, LOADING, LOADING_PATH))
 
         # Move to imaging position
         cryoSwitchSamplePosition(LOADING).result()
```

Comment:
Is it not necessary to check whether the CurrentPositionLabel correctly switched to LOADING?

Reply:
We already do that line 250. It's not really needed to check again.